halo/app/upgrades/uluwatu/upgrade.go

```diff
@@ -8,38 +8,91 @@ import (
 	"time"
 
 	"github.com/omni-network/omni/lib/errors"
+	"github.com/omni-network/omni/lib/feature"
 
 	"cosmossdk.io/math"
 	storetypes "cosmossdk.io/store/types"
 	upgradetypes "cosmossdk.io/x/upgrade/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/cosmos/cosmos-sdk/types/module"
+	authkeeper "github.com/cosmos/cosmos-sdk/x/auth/keeper"
+	mintkeeper "github.com/cosmos/cosmos-sdk/x/mint/keeper"
+	minttypes "github.com/cosmos/cosmos-sdk/x/mint/types"
 	slkeeper "github.com/cosmos/cosmos-sdk/x/slashing/keeper"
 	sltypes "github.com/cosmos/cosmos-sdk/x/slashing/types"
 )
 
 const UpgradeName = "1_uluwatu"
 
-var StoreUpgrades storetypes.StoreUpgrades // Zero store upgrades
+func StoreUpgrades(ctx context.Context) *storetypes.StoreUpgrades {
+	var storeUpgrades storetypes.StoreUpgrades // Zero store upgrades
 
-// SlashingParams as per https://github.com/omni-network/omni/issues/2018.
-var SlashingParams = sltypes.Params{
-	SignedBlocksWindow:      2000,                            // Roughly 40min
-	MinSignedPerWindow:      math.LegacyNewDecWithPrec(5, 1), // Unchanged from default 5%
-	DowntimeJailDuration:    12 * time.Hour,                  // Mimics post-valsync-buffer network upgrade.
-	SlashFractionDoubleSign: math.LegacyNewDec(0),            // 0% since mainnet V1 has trusted operators.
-	SlashFractionDowntime:   math.LegacyNewDec(0),            // 0% since mainnet V1 has trusted operators.
+	if feature.FlagEVMStakingModule.Enabled(ctx) {
+		storeUpgrades.Added = []string{minttypes.StoreKey} // Unless staking module flag in which case we add the mint module
+	}
+
+	return &storeUpgrades
 }
 
+var (
+	// SlashingParams as per https://github.com/omni-network/omni/issues/2018.
+	SlashingParams = sltypes.Params{
+		SignedBlocksWindow:      2000,                            // Roughly 40min
+		MinSignedPerWindow:      math.LegacyNewDecWithPrec(5, 1), // Unchanged from default 5%
+		DowntimeJailDuration:    12 * time.Hour,                  // Mimics post-valsync-buffer network upgrade.
+		SlashFractionDoubleSign: math.LegacyNewDec(0),            // 0% since mainnet V1 has trusted operators.
+		SlashFractionDowntime:   math.LegacyNewDec(0),            // 0% since mainnet V1 has trusted operators.
+	}
+
+	targetInflation  = math.LegacyNewDecWithPrec(11, 0)
```

Comment:
this should be `math.LegacyNewDecWithPrec(11, 2)` I think. Also maybe add a comment `// 11%`